src/plugins/kibana_react/public/util/use_observable.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { useEffect, useState } from 'react';
+import { Observable } from 'rxjs';
+
+export function useObservable<T>(observable$: Observable<T>): T | undefined;
```

Comment:
I would really like to see this code removed from x-pack in the same PR and used from its new location.
It will be a great example.
I guess we'd have to talk to someone from ES UI team, but I don't believe it should be a problem.

Reply:
I really want to make this PR just focused on creating the library. Maybe one more PR to add readme and missing functionality needed to move `infra` plugin to this library. And only then actually start using it in `infra` plugin.

Reply:
Cool, but I would definitely make this a part of this effort, and not to leave code behind :)